### PR TITLE
Fix reported Snkyk issues - master

### PR DESCRIPTION
**Pull request by PeterSuna**
Bump versions in lighty-gnmi-proto
 protobuf 3.11.3 -> 3.19.3
 grpc 1.27.0 -> 1.44.0

Use Alpine instead of Debian in docker container
 openjdk:11-jre-slim use Debian OS which has known vulnerabilities
https://security-tracker.debian.org/tracker/CVE-2022-23218
https://security-tracker.debian.org/tracker/CVE-2022-23219
https://security-tracker.debian.org/tracker/CVE-2021-33574
https://security-tracker.debian.org/tracker/CVE-2020-16156
https://security-tracker.debian.org/tracker/CVE-2021-33560

 Hence, we changed it to Alpine Linux and install JRE inside docker.
Before docker size with openjdk:11-jre-slim:
   - RNC 15.1.0    : 352 MB
   - RcGNMI 15.1.0 : 337 MB

Size of docker images with Alpine:
   - RNC 15.1.0    : 311 MB
   - RcGNMI 15.1.0 : 296 MB
   
Fix lighty applications helm-charts vulnerability:
https://snyk.io/security-rules/SNYK-CC-K8S-6
https://snyk.io/security-rules/SNYK-CC-K8S-9
https://snyk.io/security-rules/SNYK-CC-K8S-10

### CODE REVIEW

**Review by ihrasko — APPROVED** [review]

File: lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-app-helm/helm/lighty-rnc-app-helm/values.yaml [hunk 1]
```diff
@@ -69,6 +69,14 @@ ingress:
   exposeManagement: false
   managementHost: "management.lighty.io"
 
+deploymentSecurity:
+  usePodSecurity: true
+  useContainerSecurity: true
+  # User with UID 1000 is set in RNC docker image
+  runAsUser: 1000
+  # Description for possible capabilities https://github.com/torvalds/linux/blob/master/include/uapi/linux/capability.h
+  useCapabilities: ["NET_BIND_SERVICE"]
```
Comment (ihrasko) [inline lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-app-helm/helm/lighty-rnc-app-helm/values.yaml:78]: We are just binding 8888 - why we need capability?

File: lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-app-helm/helm/lighty-rnc-app-helm/templates/deployment.yaml [hunk 2]
```diff
@@ -25,6 +30,15 @@ spec:
               mountPath: {{ .Values.lighty.workdir }}/{{ .Values.lighty.configDirectoryName }}
             - name: secrets-volume
               mountPath: {{ .Values.lighty.workdir }}/{{ .Values.lighty.restconf.keyStoreDirectory }}
+          {{- if .Values.deploymentSecurity.useContainerSecurity }}
+          securityContext:
+            allowPrivilegeEscalation: true
+            runAsUser: {{ .Values.deploymentSecurity.runAsUser }}
+            capabilities:
```
Comment (ihrasko) [inline lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-app-helm/helm/lighty-rnc-app-helm/templates/deployment.yaml:37]: Why?

File: lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-app-helm/helm/lighty-rnc-app-helm/templates/deployment.yaml [hunk 3]
```diff
@@ -25,6 +30,15 @@ spec:
               mountPath: {{ .Values.lighty.workdir }}/{{ .Values.lighty.configDirectoryName }}
             - name: secrets-volume
               mountPath: {{ .Values.lighty.workdir }}/{{ .Values.lighty.restconf.keyStoreDirectory }}
+          {{- if .Values.deploymentSecurity.useContainerSecurity }}
+          securityContext:
+            allowPrivilegeEscalation: true
```
Comment (ihrasko) [inline lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-app-helm/helm/lighty-rnc-app-helm/templates/deployment.yaml:35]: Setting allowPrivilegeEscalation=false is strongly recommended with this strategy.

https://kubernetes.io/docs/concepts/policy/pod-security-policy/

File: lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-app-helm/helm/lighty-rnc-app-helm/values.yaml [hunk 4]
```diff
@@ -69,6 +69,14 @@ ingress:
   exposeManagement: false
   managementHost: "management.lighty.io"
 
+deploymentSecurity:
+  usePodSecurity: true
```
Comment (ihrasko) [inline lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-app-helm/helm/lighty-rnc-app-helm/values.yaml:73]: Do we have use case when we do not want use "security"?

**Review by ihrasko — CHANGES_REQUESTED** [review]

**Review by ihrasko — APPROVED** [review]

File: lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-app-docker/src/main/docker/Dockerfile [hunk 5]
```diff
@@ -1,6 +1,17 @@
-FROM openjdk:11-jre-slim
+FROM alpine:latest
+
+RUN apk add openjdk11-jre
 
 WORKDIR /lighty-rnc
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-app-docker/src/main/docker/Dockerfile:5]: done

File: lighty-applications/lighty-rcgnmi-app-aggregator/lighty-rcgnmi-app-docker/src/main/docker/Dockerfile [hunk 6]
```diff
@@ -1,6 +1,17 @@
-FROM openjdk:11-jre-slim
+FROM alpine:latest
+
+RUN apk add openjdk11-jre
 
 WORKDIR /lighty-rcgnmi
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply lighty-applications/lighty-rcgnmi-app-aggregator/lighty-rcgnmi-app-docker/src/main/docker/Dockerfile:5]: done

File: lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-app-helm/helm/lighty-rnc-app-helm/values.yaml (hunk 4 above)
Comment (PeterSuna) [reply lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-app-helm/helm/lighty-rnc-app-helm/values.yaml]: Removed optional security for deployment

File: lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-app-helm/helm/lighty-rnc-app-helm/templates/deployment.yaml (hunk 3 above)
Comment (PeterSuna) [reply lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-app-helm/helm/lighty-rnc-app-helm/templates/deployment.yaml]: Fixed

File: lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-app-helm/helm/lighty-rnc-app-helm/values.yaml (hunk 1 above)
Comment (PeterSuna) [reply lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-app-helm/helm/lighty-rnc-app-helm/values.yaml]: Removed

**Pull request merged**